### ISSUE TITLE
Destructuring context help broken

### ISSUE THREAD

**Issue by cemoktra** [zed-industries/zed#18380]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Since the last update the auto "fill struct fields" does not work anymore on Self destructing:
```rust
let Self { } = self;
```

Pressing CTRL + space here does not give "fill struct fields"

### Environment

Zed: v0.154.2 (Zed)
OS: Linux Wayland ubuntu 24.04
Memory: 31.1 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics 630 (CFL GT2) || Intel open-source Mesa driver || Mesa 24.0.5-1ubuntu1

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/user-attachments/assets/0513a2bf-701d-4c5b-ba63-ab1c85b056b4)

"implement default" refers that i'm trying to impl a trait that has default implementation. Changing to explicit type does not help here.

**Comment by maxdeviant** [comment]
Does anything change if you replace `Self` with the explicit type first?

**Comment by cemoktra** [comment]
> Does anything change if you replace `Self` with the explicit type first?

No. Also it does not only affect Self i now noticed. It is about all destructuring

**Comment by SomeoneToIgnore** [comment]
I've tried a simplistic example and that seems to work both in VSCode and Zed:

https://github.com/user-attachments/assets/f5375fd3-ede4-4609-bc26-30185dc77de0


https://github.com/user-attachments/assets/9c747abf-4971-428c-b7f2-dd444b53b2b1

I wonder, whether this all could be caused by the fact that latest Zed now tries to autodetect and use r-a from the path.
Please check your logs for lines like
```
   26 2024-09-26T19:43:42.84573+03:00 [INFO] starting language server process. binary path: "/Users/someonetoignore/Library/Application Support/Zed/languages/rust-analyzer/rust-analyzer-2024-09-16", working directory: "/Users/someonetoignore/work/zed/zed", args: []¬
```

to understand what it tries to start.

To ignore any local versions and make Zed to download the latest stable r-a always, use `"ignore_system_version": true` as described in https://zed.dev/docs/languages/rust#binary

**Comment by cemoktra** [comment]
For me it makes no difference which rust analyzer i use

**Comment by SomeoneToIgnore** [comment]
Closing as stale, I cannot reproduce it still.

**Comment by cemoktra** [comment]
https://github.com/user-attachments/assets/92c994c9-8dfd-4979-9a37-3af52bd26cd2

**Comment by cemoktra** [comment]
Same applies to "fill match arms"

**Comment by cemoktra** [comment]
The `.match` postfix does work tho

**Comment by cemoktra** [comment]
I also tried to disable all custom options and reset my settings.json to `{}` without any change

**Comment by SomeoneToIgnore** [comment]
I have no idea what's broken in your case, commenting out settings and keeping them yield the same result for me — things work.

Here's a video with both settings commented and not, using Stable:

https://github.com/user-attachments/assets/d4aa1724-a248-4f51-831d-15a05ac3d726

Notice the `open language server logs` command that I call and open `Server Info` tab there — inside, a full path to rust-analyzer used is shown.
You can try doing the same to ensure it's not some stale version you're picking up.

**Comment by SomeoneToIgnore** [comment]
One thing though: in your latest video, you seem to trigger completions (judging by the menu appearance) and not actions (as in your screenshot in the OP)?

That was never a thing for rust-analyzer, filling struct fields should happen via actions (via cmd-. on mac or that lightning icon in the gutter pressed).

**Comment by cemoktra** [comment]
> One thing though: in your latest video, you seem to trigger completions (judging by the menu appearance) and not actions (as in your screenshot in the OP)?
> 
> That was never a thing for rust-analyzer, filling struct fields should happen via actions (via cmd-. on mac or that lightning icon in the gutter pressed).

What, yeah there it is. I'm new to Mac, on Linux it was the same menu tho.
